api/institutions/views.py

```diff
@@ -29,12 +36,12 @@ class InstitutionMixin(object):
 
     def get_institution(self):
         inst = get_object_or_error(
-            Node,
-            Q('institution_id', 'eq', self.kwargs[self.institution_lookup_url_kwarg]),
+            Institution,
+            self.kwargs[self.institution_lookup_url_kwarg],
             display_name='institution',
             allow_institution=True
```

Comment:
The `Institution.find()` override automatically includes `allow_institution` in a `Node.find` call; you should be able to remove this line.